jscomp/bsb/bsb_pkg.ml

```diff
@@ -101,7 +101,65 @@ let resolve_bs_package ~cwd (package : t) =
       end;
     x
 
+(* Some package managers will implement "postinstall" caches, that do not
+ * keep their build artifacts in the local node_modules. Similar to
+ * npm_config_prefix, bs_custom_resolution allows these to specify the
+ * exact location of build cache, but on a per-package basis. Implemented as
+ * environment lookup to avoid invasive changes to bsconfig and mandates. *)
+let custom_resolution =
+ match Sys.getenv "bs_custom_resolution" with
+  | exception Not_found  -> false
+  | "true"  -> true
+  | _ -> false
 
+let regex_at = Str.regexp "@"
+let regex_unders = Str.regexp "_+"
+let regex_slash = Str.regexp "\\/"
+let regex_dot = Str.regexp "\\."
+let regex_hyphen = Str.regexp "-"
+let pkg_name_as_variable pkg =
+  Bsb_pkg_types.to_string pkg
+  |> Str.replace_first regex_at ""
+  |> Str.global_replace regex_unders "\\0_"
+  |> Str.global_replace regex_slash "__slash__"
+  |> Str.global_replace regex_dot "__dot__"
+  |> Str.global_replace regex_hyphen "_"
+
+let resolve_bs_package ~cwd (pkg : t) =
+  if custom_resolution then
+  begin
```

Comment:
can you change the current resolve_bs_package instead of shadowing, it is hard to review what changes are made

Reply:
Moved it to be in the else case